vermouth/tests/test_geometry.py

```diff
@@ -0,0 +1,129 @@
+# -*- coding: utf-8 -*-
+# Copyright 2018 University of Groningen
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Tests for the geometry module.
+"""
+
+import pytest
+import numpy as np
+from vermouth import geometry
+
+
+def _generate_test_angles(n_angles):
+    """
+    Gererate a series of coordinates of 3 points at a different angles.
+
+    Generate 'n_angles' structures with angles between 0 and pi with regular
+    angle spacing.
+
+    Parameters
+    ----------
+    n_angles: int
+        Number of structures to generate.
+
+    Yields
+    ------
+    coordinates: np.ndrarray
+        The coordinates of the points. Each row corresponds to a pointm and
+        each column corresponds to a dimension.
+    angle: float
+        The angle between the 3 points.
+    """
+    # The shift is arbitrary. Its purpose is to avoid having the angle centered
+    # at the origin as it may be a special case.
```

Comment:
So make sure we also test the special case ;)

Reply:
I understand that. By the way, are you sure the broadcasting of (3, 3) + (3, ) happens as you expect it to? Not that I really matters if it doesn't.